### ISSUE TITLE
RainAlarm App by mdiener shows distorted gmap after screen rotate

### ISSUE THREAD

**Issue by Catfriend1** [microg/android_packages_apps_GmsCore#147]
Hi

This is a Problem i believe being present since a long time ago. Normally I habe my screen locked to portrait for most apps. Today I enabled auto rotate in Android and hold it so it goes to landscape. So I just noticed the probably old Problem.

Then it looks like this: https://postimg.org/image/ez9lxqvzr/

Link to App: https://play.google.com/store/apps/details?id=de.mdiener.rain.usa

System : cm13 nightly 20160529, LG G4-H815 mobile, MicroG 0.2.3-1 GmsCore, Xposed FakeGApps

**Comment by mar-v-in** [comment]
[vtm](https://github.com/opensciencemap/vtm), the library we use to render the map unfortunately was discontinued. It still has quiet some number of annoying bugs, especially when the map view is recreated (which happens on rotation), the map contains a lot of elements or when multiple map views are visible at the same time. I am not very deep into OpenGL 3d programming and can't do a lot on that.

The designated successor is [tangram](https://github.com/tangrams/tangram-es) or its wrapper [mapzen-android-sdk](https://github.com/mapzen/android), however it still lacks a lot of features, most importantly proper markup (markers/lines/etc) support. Until that, I think it's the best to just keep what we have. Rendering problems are very unfortunate, but often it's possible to "fix" them by restarting the app.

**Comment by lambdaupb** [comment]
The mapzen android SDK claims support for polylines and markers on https://github.com/mapzen/android (docs added on Apr 21 2016)

**Comment by mar-v-in** [comment]
@lambdaupb yes, but no support of coloring, input listeners etc. It's currently only drawing a line on the map.

**Comment by ghost** [comment]
please delete issue
